src/MicrosoftTeams.ts

```diff
@@ -417,25 +442,132 @@ namespace microsoftTeams
         {
             ensureInitialized(frameContexts.content, frameContexts.settings, frameContexts.remove);
 
+            // Open an authentication window with the parameters provided by the caller
+            openAuthenticationWindow(authenticateParameters.url, authenticateParameters.width, authenticateParameters.height);
+
+            authParams = authenticateParameters;
+        }
+
+        function closeAuthenticationWindow(): void
+        {
+            // Stop monitoring the authentication window
+            stopAuthenticationWindowMonitor();
+
+            // Try to close the authentication window and clear all properties associated with it
+            try
+            {
+                if (childWindow)
+                {
+                    childWindow.close();
+                }
```

Comment:
Well in certain cases (e.g. where authenticate gets called again before the previous auth flow completes) we would need to be the ones to close the popup. Since closeAuthenticationWindow does more than just close the window it needs to be called in several places where a call to childWindow.close() isn't necessarily needed; however, I think it's simplest and safest to just do it in all cases. Worst case the browser gets two requests to close the popup which should be just fine.